File: Examples/VisualBasic/Programming-Documents/Document/DocumentBuilderInsertParagraph.vb
Imports System.IO
Imports Aspose.Words
Imports System.Drawing

Class DocumentBuilderInsertParagraph
    Public Shared Sub Run()
        ' ExStart:DocumentBuilderInsertParagraph
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_WorkingWithDocument()
        ' Initialize document.
        Dim doc As New Document()
        Dim builder As New DocumentBuilder(doc)

        ' Specify font formatting
        Dim font As Aspose.Words.Font = builder.Font
        font.Size = 16
        font.Bold = True
        font.Color = System.Drawing.Color.Blue
        font.Name = "Arial"
        font.Underline = Underline.Dash

        ' Specify paragraph formatting
        Dim paragraphFormat As ParagraphFormat = builder.ParagraphFormat
        paragraphFormat.FirstLineIndent = 8
        paragraphFormat.Alignment = ParagraphAlignment.Justify
        paragraphFormat.KeepTogether = True

        builder.Writeln("A whole paragraph.")
        dataDir = dataDir & Convert.ToString("DocumentBuilderInsertParagraph_out_.doc")
        doc.Save(dataDir)
        ' ExEnd:DocumentBuilderInsertParagraph
        Console.WriteLine(Convert.ToString(vbLf & "Paragraph inserted successfully into the document using DocumentBuilder." & vbLf & "File saved at ") & dataDir)
    End Sub
End Class

File: Examples/CSharp/Programming-Documents/Document/DocumentBuilderInsertParagraph.cs
using System.IO;
using Aspose.Words;
using System;
using System.Drawing;
namespace Aspose.Words.Examples.CSharp.Programming_Documents.Working_With_Document
{
    class DocumentBuilderInsertParagraph
    {
        public static void Run()
        {
            //ExStart:DocumentBuilderInsertParagraph
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WorkingWithDocument();
            // Initialize document.
            Document doc = new Document();
            DocumentBuilder builder = new DocumentBuilder(doc);

            // Specify font formatting
            Font font = builder.Font;
            font.Size = 16;
            font.Bold = true;
            font.Color = System.Drawing.Color.Blue;
            font.Name = "Arial";
            font.Underline = Underline.Dash;

            // Specify paragraph formatting
            ParagraphFormat paragraphFormat = builder.ParagraphFormat;
            paragraphFormat.FirstLineIndent = 8;
            paragraphFormat.Alignment = ParagraphAlignment.Justify;
            paragraphFormat.KeepTogether = true;

            builder.Writeln("A whole paragraph.");
            dataDir = dataDir + "DocumentBuilderInsertParagraph_out_.doc";
            doc.Save(dataDir);
            //ExEnd:DocumentBuilderInsertParagraph
            Console.WriteLine("\nParagraph inserted successfully into the document using DocumentBuilder.\nFile saved at " + dataDir);
        }        
    }
}
